src/App.tsx

```diff
@@ -1,223 +1,187 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { getFilteredTodos } from './api/helper';
+import { addTodo, getTodos, deleteTodo } from './api/todos';
+import { AppFooter } from './components/AppFooter/AppFooter';
+import { AppHeader } from './components/AppHeader/AppHeader';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification }
+  from './components/ErrorNotification/ErrorNotification';
+import { TodoList } from './components/TodoList/TodoList';
+import { Filter } from './types/Filter';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [statusFilter, setStatusFilter] = useState<Filter>('All');
+  const [error, setError] = useState('');
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isAdding, setIsAdding] = useState(false);
+  const [loadingTodosIds, setLoadingTodosIds] = useState<number[]>([]);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    if (user) {
+      try {
+        getTodos(user.id)
+          .then(setTodos);
+      } catch {
+        setError('Can\'t load todo');
+      }
+    }
+  }, []);
+
+  const filteredTodos = useMemo(() => (
+    getFilteredTodos(todos, statusFilter)
+  ), [todos, statusFilter]);
+
+  const amountOfActive = useMemo(() => (
+    todos.filter(
+      todo => !todo.completed,
+    ).length
+  ), [todos]);
+
+  const completedTodosLength = useMemo(() => (
+    todos.filter(
+      todo => todo.completed,
+    ).length
+  ), [todos]);
+
+  const isAllCompleted = completedTodosLength === todos.length;
+
+  const addingNewTodo = (event: React.FormEvent) => {
```

Comment:
We passing this function to the another component, so you should wrap it with `useCallback`